core/gui/src/app/workspace/component/workspace.component.ts

```diff
@@ -119,16 +120,28 @@ export class WorkspaceComponent implements AfterViewInit, OnInit, OnDestroy {
   }
 
   ngOnDestroy() {
+    this.saveLatestWorkflow();
     this.workflowActionService.setWorkflowMetadata(undefined);
     this.workflowActionService.destroySharedModel();
     this.workflowWebsocketService.closeWebsocket();
   }
 
   @HostListener("window:beforeunload")
   unloadHandler() {
+    this.saveLatestWorkflow();
     this.vc.clear();
```

Comment:
also, just curious, why does workspace has a reference to a code editor, and manages its life cycle? ask @aglinxinyuan for context?

Reply:
Workspace is the parent of code editor floating panels.